pytest_fixture_classes/fixture_class.py

```diff
@@ -58,19 +65,67 @@ def inner(fixture_cls):
             fixture_cls
         )
         args = list(inspect.signature(fixture_dataclass.__init__).parameters)[1:]
-        func_def = dedent(
-            f"""
-            def {fixture_dataclass.__name__}({', '.join(args)}):
-                return fixture_cls({', '.join(args)})
-        """
-        )
+
+        has_lifespan = hasattr(fixture_cls, "lifespan")
+        is_async_lifespan = False
+
+        if has_lifespan:
```

Comment:
How about just inserting a lifespan that does nothing in case there is no lifespan? It should make the code much simpler and less duplicated